test/statistex_test.exs

```diff
@@ -12,6 +12,104 @@ defmodule Statistex.StatistexTest do
+                 mode: [500, 400],
+                 minimum: 200,
+                 maximum: 900,
+                 outliers_bounds: {200, 900.0},
+                 outliers: [],
+                 sample_size: 9
+               }
+    end
+
+    test "returns Statistex struct with outliers" do
+      assert Statistex.statistics([50, 50, 450, 450, 450, 500, 500, 500, 600, 900]) ==
+               %Statistex{
+                 total: 4450,
+                 average: 445.0,
+                 variance: 61361.11111111111,
+                 standard_deviation: 247.71175004652304,
+                 standard_deviation_ratio: 0.5566556180820742,
+                 median: 475.0,
+                 percentiles: %{25 => 350.0, 50 => 475.0, 75 => 525.0},
+                 frequency_distribution: %{50 => 2, 450 => 3, 500 => 3, 600 => 1, 900 => 1},
+                 mode: [500, 450],
+                 minimum: 50,
+                 maximum: 900,
+                 outliers_bounds: {87.5, 787.5},
+                 outliers: [50, 50, 900],
+                 sample_size: 10
+               }
+    end
+
+    test "returns Statistex struct with excluded outliers once" do
+      assert Statistex.statistics([50, 50, 450, 450, 450, 500, 500, 500, 600, 900],
+               exclude_outliers: :once
+             ) ==
+               %Statistex{
+                 total: 3450,
+                 average: 492.85714285714283,
+                 variance: 2857.142857142857,
+                 standard_deviation: 53.452248382484875,
+                 standard_deviation_ratio: 0.1084538372977954,
+                 median: 500.0,
+                 percentiles: %{25 => 450.0, 50 => 500.0, 75 => 500.0},
+                 frequency_distribution: %{450 => 3, 500 => 3, 600 => 1},
+                 mode: [500, 450],
+                 minimum: 450,
+                 maximum: 600,
+                 outliers_bounds: {450, 575.0},
+                 outliers: [600, 50, 50, 900],
+                 sample_size: 7
+               }
+    end
+
+    test "returns Statistex struct with excluded outliers repeatedly" do
+      assert Statistex.statistics([50, 50, 450, 450, 450, 500, 500, 500, 600, 900],
+               exclude_outliers: :repeatedly
+             ) ==
+               %Statistex{
+                 total: 2850,
+                 average: 475.0,
+                 variance: 750.0,
+                 standard_deviation: 27.386127875258307,
+                 standard_deviation_ratio: 0.05765500605317538,
+                 median: 475.0,
+                 percentiles: %{25 => 450.0, 50 => 475.0, 75 => 500.0},
+                 frequency_distribution: %{450 => 3, 500 => 3},
+                 mode: [500, 450],
+                 minimum: 450,
+                 maximum: 500,
+                 outliers_bounds: {450, 500},
+                 outliers: [50, 50, 900, 600],
+                 sample_size: 6
+               }
+    end
+  end
+
```

Comment:
I'd love it if we could add some examples from some online text books or something where they're discussing outlier detection with the method we use. Just to have these in the test base to make sure we're "working as according to text book" according to at least some sources :sweat_smile: 

But also don't worry about it, I can add that.